### DISCUSSION TITLE
Script works fine in terminal, but not when lauched from config.

### DISCUSSION THREAD

**Discussion by Codester02** [i3/i3#6261] [Q&A] [answered]
I'm working on setting up a wallpaper changer that randomly sets my wallpaper from a folder. The script works perfectly when launched from terminal, but when i try to add it to my config using exec_always or bindsym it doesn't work.

Script:
```
#1/bin/bash
files=(~/Pictures/Backgrounds/*)
while [ 1 ]; do
	randompic=`printf "%s\n" "${files[RANDOM % ${#files[@]}]}"`

	wal -i "$randompic"
	sleep 1
done
```

My searching got me to a Reddit post that told me to check .xsession-errors log which gave this error when I launched the script from config:
`/home/codef/termwal.sh: 2: Syntax error: "(" unexpected`

**Comment by okraits** [comment, marked as answer]
The shebang in your script is not correct. Or is that just a copy-paste error?

Please also make sure then that you use correct bash syntax.

**Reply by Codester02** [reply]
Yep I totally missed the shift key and didn't notice. Works perfectly now.
